http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -439,9 +446,65 @@ class DataprocInterpreter[F[_]: Timer: Parallel: ContextShift](
   def removeClusterIamRoles(googleProject: GoogleProject, serviceAccountInfo: WorkbenchEmail): F[Unit] =
     updateClusterIamRoles(googleProject, serviceAccountInfo, createCluster = false)
 
+  // Called at Leo boot time. Sets up the Dataproc image user group if it does not already exist.
+  // Group membership is modified at Dataproc cluster creation time.
   def setupDataprocImageGoogleGroup(implicit ev: Ask[F, AppContext]): F[Unit] =
-    createDataprocImageUserGoogleGroupIfItDoesntExist >>
-      addIamRoleToDataprocImageGroup
+    for {
+      ctx <- ev.ask
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Checking if Dataproc image user Google group '${config.groupsConfig.dataprocImageProjectGroupEmail}' already exists..."
+      )
+      // Check if the group exists
+      groupOpt <- F.liftIO(
+        IO.fromFuture[Option[Group]](
+          IO(googleDirectoryDAO.getGoogleGroup(config.groupsConfig.dataprocImageProjectGroupEmail))
+        )
+      )
+      // Create the group if it does not exist
+      _ <- groupOpt match {
+        case None =>
+          F.liftIO(
+              IO.fromFuture(
+                IO(
+                  googleDirectoryDAO
+                    .createGroup(
+                      config.groupsConfig.dataprocImageProjectGroupName,
+                      config.groupsConfig.dataprocImageProjectGroupEmail,
+                      Option(googleDirectoryDAO.lockedDownGroupSettings)
+                    )
+                )
+              )
+            )
+            .handleErrorWith {
+              case t if when409(t) => F.unit
+              case t =>
+                F.raiseError(
+                  GoogleGroupCreationException(config.groupsConfig.dataprocImageProjectGroupEmail, t.getMessage)
+                )
+            }
+        case Some(group) =>
+          logger.info(ctx.loggingCtx)(
+            s"Dataproc image user Google group '${config.groupsConfig.dataprocImageProjectGroupEmail}' already exists: $group \n Won't attempt to create it."
+          )
+      }
+      // Add compute.imageUser role to the group in the custom image's project
+      imageProject <- F.fromOption(parseImageProject(config.dataprocConfig.customDataprocImage),
+                                   ImageProjectNotFoundException)
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Attempting to grant 'compute.imageUser' permissions to '${config.groupsConfig.dataprocImageProjectGroupEmail}' on project '$imageProject' ..."
+      )
+      _ <- retry(
```

Comment:
does this call need to be inside `retry`?

Reply:
This call happens at boot time so probably not.. also probably doesn't hurt though